File: Programmers/Level2/오픈채팅방/오픈채팅방.py
def solution(record):
    answer = []
    user_dict = {}
    
    for r in record:
        r_split = r.split()
        if len(r_split) == 3:
            user_dict[r_split[1]] = r_split[2]
    
    for r in record:
        r_split = r.split()
        if r_split[0] == "Enter":
            answer.append("{}님이 들어왔습니다.".format(user_dict[r_split[1]]))
        elif r_split[0] == "Leave":
            answer.append("{}님이 나갔습니다.".format(user_dict[r_split[1]]))
            
    return answer

File: Programmers/Level2/오픈채팅방/오픈채팅방.js
function solution(record) {
    let answer = [];
    _map = new Map();

    for (let r of record) {
        _split = r.split(" ");

        _split.length === 3 && _map.set(_split[1], _split[2]);
    }

    for (let r of record) {
        _split = r.split(" ");
        if (_split[0] == "Enter") {
            answer.push(`${_map.get(_split[1])}님이 들어왔습니다.`);
        } else if (_split[0] == "Leave") {
            answer.push(`${_map.get(_split[1])}님이 나갔습니다.`);
        }
    }

    return answer;
}
